### ISSUE TITLE
Custom Component Retrieval Issue in Build (getComponent returns null)

### ISSUE THREAD

**Issue by tafedorov78** [cocos/cocos-engine#17898]
### Cocos Creator version

3.8.3

### System information

chrome, macos, windows

### Issue description

I’m experiencing an issue with a custom component attached to a node. When accessing the component locally using the getComponent method, it returns the expected component. However, when accessing the component remotely in a build from remote host, it returns null.
in the attached project - look at PrizeController -->             const switcher2 = node.getComponent('NodeSwitchComponent');

### Relevant error log output

null

### Steps to reproduce

1.	Attach a custom component to a node.
2.	Access the component locally using getComponent - the component is retrieved correctly.
3.	Access the component remotely (remote host!!!) in a build - getComponent returns null.

### Minimal reproduction project

[test.zip](https://github.com/user-attachments/files/17835519/test.zip)

**Comment by longchuan** [comment]
![image](https://github.com/user-attachments/assets/7c2cb346-30ee-4ade-b49f-e7ab27e66724)
look at this

**Comment by longchuan** [comment]
![image](https://github.com/user-attachments/assets/2a327f78-3328-46e2-8d23-a548d851f538)
Thanks for I got a gift program😂

**Comment by tafedorov78** [comment]
I don't think you got it. I mentioned it should be run it remotely, not on your local machine that the point.

**Comment by tafedorov78** [comment]
I found one thing - if I set a target in tsconfig = es6 it does not work!

**Comment by longchuan** [comment]
> I don't think you got it. I mentioned it should be run it remotely, not on your local machine that the point.

A slowly network: http://119.3.232.55/web-mobile/index.html
![image](https://github.com/user-attachments/assets/5ab9ece1-041f-4036-a117-c28a8707c4ca)

**Comment by longchuan** [comment]
The root path of import is the dir of script file.So,you should add "../" to return parent dir.

**Comment by minggo** [comment]
> Access the component remotely (remote host!!!) in a build - getComponent returns null.

How can you access remote component? I don't think it is supported.
